website/blog/2024-06-18-monitor-ferretdb-performance-using-coroot.md

```diff
@@ -0,0 +1,134 @@
+
+Effective real-time monitoring is a critical aspect of any infrastructure.
+[Coroot](https://coroot.com/) is an open source observability platform that can provide real-time monitoring and visibility into a [FerretDB](https://www.ferretdb.com/) setup.
+
+<!--truncate-->
+
+App developers can quickly troubleshoot issues without necessarily becoming experts in infrastructure operations and management.
+
+If you're just finding out about FerretDB, it's the truly open source document database that provides MongoDB compatibility with Postgres as the backend.
+With Coroot, you can get visibility into your entire system in a matter of minutes.
+
+In this blog post, you'll learn about Coroot and how it can help you monitor and get better visibility into your FerretDB instance.
+
+## Coroot observability for FerretDB
+
+Imagine a scenario where your FerretDB instance is deployed as a Docker container along with other components and services for your application.
+Observability into all critical areas of the system in as little time is vital.
+However, it can be challenging to monitor all components and services effectively.
+You need to instrument all parts of your application, including the FerretDB, PostgreSQL, and other services and orchestrators connected to it.
+That could be resource-intensive, time-consuming, and even costly to configure, log, and maintain.
+
+With Coroot, you can visualize every aspect of your entire setup.
+That includes all its components, including your applications and their subsystems, databases, and, most importantly, how they all communicate.
+
+Coroot heavily relies on eBPF, a technology that enhances observability and monitoring by eliminating the need to instrument application code manually.
+Custom programs can run directly within the Linux kernel without changing kernel source code or loading additional kernel modules.
+You get deeper and more extensive information on system calls, network functions, and security issues – just better observability.
+
+Let's look at an example with Prometheus and Grafana.
+
+### Monitoring FerretDB with Prometheus and Grafana
+
+A typical approach to monitoring your FerretDB instances is through Prometheus and Grafana.
+You can deploy Prometheus to scrape metrics from FerretDB and visualize them through Grafana dashboards.
+That process involves:
+
+- First, you need to expose the default port for FerretDB metrics in Prometheus format on http://127.0.0.1:8088/debug/metrics.
+  There's no need for an external exporter.
+
+- Next, you need to set up Prometheus for metrics collection
+
+- Then go ahead to design all the dashboards and panels to visualize.
+
+The image below shows a typical dashboard design using Grafana to display some of the metrics, including total client requests and responses, client connection durations, memory usage, CPU usage, and overall instance health.
+
+![Grafana dashboard for FerretDB Prometheus metrics 1](/img/blog/ferretdb-coroot/grafana-prometheus.png)
```

Comment:
The image for Grafana is just to show the dashboard setup rather than specific metrics. Others are more specific. But yes, this is something to look at actually. We may enable images to be clickable/zoomable via some plugin addition to Docusuaurus.